crates/catalog/rest/src/catalog.rs

```diff
@@ -0,0 +1,845 @@
+        ))
+    }
+
+    /// Drop a namespace from the catalog.
+    async fn drop_namespace(&self, namespace: &NamespaceIdent) -> Result<()> {
+        let request = self
+            .client
+            .0
+            .delete(self.config.namespace_endpoint(namespace))
+            .build()?;
+
+        self.client.execute::<ErrorModel, NO_CONTENT>(request).await
+    }
+
+    /// List tables from namespace.
+    async fn list_tables(&self, namespace: &NamespaceIdent) -> Result<Vec<TableIdent>> {
+        let request = self
+            .client
+            .0
+            .get(self.config.tables_endpoint(namespace))
+            .build()?;
+
+        let resp = self
+            .client
+            .query::<ListTableResponse, ErrorModel, OK>(request)
+            .await?;
+
+        Ok(resp.identifiers)
+    }
+
+    /// Create a new table inside the namespace.
+    async fn create_table(
+        &self,
+        _namespace: &NamespaceIdent,
+        _creation: TableCreation,
+    ) -> Result<Table> {
+        Err(Error::new(
+            ErrorKind::FeatureUnsupported,
+            "Creating table not supported yet!",
+        ))
+    }
+
+    /// Load table from the catalog.
+    async fn load_table(&self, _table: &TableIdent) -> Result<Table> {
+        Err(Error::new(
+            ErrorKind::FeatureUnsupported,
+            "Creating table not supported yet!",
+        ))
+    }
+
+    /// Drop a table from the catalog.
+    async fn drop_table(&self, table: &TableIdent) -> Result<()> {
+        let request = self
+            .client
+            .0
+            .delete(self.config.table_endpoint(table))
+            .build()?;
+
+        self.client.execute::<ErrorModel, NO_CONTENT>(request).await
+    }
+
+    /// Check if a table exists in the catalog.
+    async fn stat_table(&self, table: &TableIdent) -> Result<bool> {
+        let request = self
+            .client
+            .0
+            .head(self.config.table_endpoint(table))
+            .build()?;
+
+        self.client
+            .execute::<ErrorModel, NO_CONTENT>(request)
+            .await
+            .map(|_| true)
+    }
+
+    /// Rename a table in the catalog.
+    async fn rename_table(&self, src: &TableIdent, dest: &TableIdent) -> Result<()> {
+        let request = self
+            .client
+            .0
+            .post(self.config.rename_table_endpoint())
+            .json(&RenameTableRequest {
+                source: src.clone(),
+                destination: dest.clone(),
+            })
+            .build()?;
+
+        self.client.execute::<ErrorModel, NO_CONTENT>(request).await
+    }
+
+    /// Update a table to the catalog.
+    async fn update_table(&self, _table: &TableIdent, _commit: TableCommit) -> Result<Table> {
+        todo!()
+    }
+
+    /// Update multiple tables to the catalog as an atomic operation.
+    async fn update_tables(&self, _tables: &[(TableIdent, TableCommit)]) -> Result<()> {
+        todo!()
+    }
```

Comment:
I'm not sure if this is the way that we want to expose update tables. It is important that the right updates and requirements are set, otherwise, race conditions might occur. Also, there is little validation here, for example, can you do two distinct schema changes (so two updates), in a single commit? I would maybe leave this out for now so we can decide later on. 

We might want to introduce a similar API as PyIceberg (which is inspired on Java): https://py.iceberg.apache.org/api/#schema-evolution

Reply:
Yes, I agree. It should be called by transaction api rather by user directly. The main challenge is that we should limit its visibility rather than exposing directly to user. I will figure out a way when actually implementing it.

Reply:
As I have said in pr description, the goal of this pr is to implement simple apis. Others will be left in following pr.

Reply:
Thank you! Just some context around the concern: Within Iceberg, when creating a new table, the IDs are being reassigned ([python](https://github.com/apache/iceberg-python/blob/0a077fcc0dceb95ca4316505bba1f41c60f8f74f/pyiceberg/schema.py#L1221-L1273), [java](https://github.com/apache/iceberg/blob/385a6bd296e551bdfdab48d2a291d3516f999f5a/api/src/main/java/org/apache/iceberg/types/AssignFreshIds.java#L27)). This can lead to confusion and potentially also errors. Therefore we refrain users from assigning IDs as much as possible (or reassigning them).

```python
tbl = table.create_table(
    Schema(
        Field(1, "id", IntegerType()),
        Field(2, "str", StringType())
    )
)

tbl.set_schema(
    Schema(
        Field(1, "id", IntegerType()),
        Field(3, "dt", DateType()),
        Field(2, "str", StringType())
    )
)
```

Because the IDs are re-assigned, it will evolve from:

```
Schema {
  1: id
  2: str
}
```

to:
```
Schema {
  1: id
  2: dt
  3: str
}
```

And this will corrupt the table since there is no valid promotion from string to date. The general consensus is that users shouldn't care about the IDs themselves, and this should not be exposed to users through APIs. The API should handle it.